static/js/Misc.jsx

```diff
@@ -1103,35 +1104,96 @@ function useHiddenButtons() {
     return [hideButtons, handleMouseOverAdminButtons];
 }
 
-const CategoryHeader =  ({children, type, data = [], edit = true,
-                            add_subcategory = true, reorder = false,
-                            add_source = false}) => {
+const CategoryHeader =  ({children, type, data = [], buttonsToDisplay = ["subcategory", "edit"]}) => {
```

Comment:
This component is doing a lot and is creating some subcomponents inside of it which might be re-rendered needlessly, so love this, I would recommend pushing them out to separate functional components:

```
// This component handles the rendering logic based on the state and props.
const CategoryHeader =  ({children, type, data = [], buttonsToDisplay = ["subcategory", "edit"]}) => {
  const [editCategory, toggleEditCategory] = useEditToggle();
  const [addCategory, toggleAddCategory] = useEditToggle();
  const [reorderCategory, toggleReorderCategory] = useEditToggle();
  const [addSource, toggleAddSource] = useEditToggle();
  const [addSection, toggleAddSection] = useEditToggle();
  const [hiddenButtons, setHiddenButtons] = useHiddenButtons(true);

  const buttonOptions = {"subcategory": ["Add sub-category", toggleAddCategory],
                          "source": ["Add a source", toggleAddSource],
                          "section": ["Add section", toggleAddSection],
                          "reorder": ["Reorder sources", toggleReorderCategory],
                          "edit": ["Edit", toggleEditCategory]};

  let wrapper = "";
  let adminButtonsSpan = null;

  if (Sefaria.is_moderator) {
    if (editCategory) {
      adminButtonsSpan = <EditCategory toggle={toggleEditCategory} type={type} data={data}/>;
    } else if (addSource) {
      adminButtonsSpan = <AddSource toggle={toggleAddSource} slug={data.slug}/>;
    } else if (addCategory) {
      adminButtonsSpan = <AddCategory toggle={toggleAddCategory} type={type} data={data}/>;
    } else if (addSection) {
      window.location = `/add/${data}`;
    } else if (reorderCategory) {
      adminButtonsSpan = <ReorderCategory toggle={toggleReorderCategory} type={type} data={data}/>;  
    } else {
      adminButtonsSpan = <AdminButtons toggle={setHiddenButtons} buttonOptions={buttonOptions} buttonsToDisplay={buttonsToDisplay}/>;
    }
  }

  return <span className={wrapper}><span onMouseEnter={() => setHiddenButtons()}>{children}</span><span>{adminButtonsSpan}</span></span>;
}

// Create separate functional components for each case.
const EditCategory = ({toggle, type, data}) => {
  // place logic here
};

const AddSource = ({toggle, slug}) => {
  // place logic here
};

const AddCategory = ({toggle, type, data}) => {
  // place logic here
};

const ReorderCategory = ({toggle, type, data}) => {
  // place logic here
};

const AdminButtons = ({toggle, buttonOptions, buttonsToDisplay}) => {
  // place logic here
};

```